frontend/src/components/Pages/MissionHistoryPage/FilterSection.tsx

```diff
@@ -0,0 +1,207 @@
+    }
+
+    const onMinEndTimeChange = (changes: ChangeEvent<HTMLInputElement>) => {
+        changeMinEndTime(changes.target.value)
+    }
+
+    const onMaxEndTimeChange = (changes: ChangeEvent<HTMLInputElement>) => {
+        changeMaxEndTime(changes.target.value)
+    }
+
+    const onClickFilterIcon = () => {
+        setIsFilteringDialogOpen(true)
+    }
+
+    const onFilterClose = () => {
+        setIsFilteringDialogOpen(false)
+    }
+
+    const onClearFilters = () => {
+        resetFilters()
+    }
+
+    return (
+        <>
+            <StyledHeader>
+                <Search
+                    value={missionName ?? ''}
+                    placeholder={TranslateText('Search for missions')}
+                    onChange={onSearchChange}
+                />
+                <Button onClick={onClickFilterIcon}>
+                    <Icon name={Icons.Filter} size={32} />
+                    {TranslateText('Filter')}
+                </Button>
+                <Button variant="outlined" onClick={onClearFilters}>
+                    <Icon name={Icons.Clear} size={32} />
+                    {TranslateText('Clear all filters')}
+                </Button>
+            </StyledHeader>
+            <Dialog open={isFilteringDialogOpen} isDismissable>
+                <StyledDialog>
+                    <StyledHeader>
+                        <Typography variant="h2">{TranslateText('Filter')}</Typography>
+                        <Button variant="ghost_icon" onClick={onFilterClose}>
+                            <Icon name={Icons.Clear} size={32} />
+                        </Button>
+                    </StyledHeader>
+                    <Autocomplete
+                        options={Object.values(MissionStatus)}
+                        onOptionsChange={onStatusesChange}
+                        label={TranslateText('Mission status')}
+                        initialSelectedOptions={statuses}
+                        multiple
+                    ></Autocomplete>
+                    <Autocomplete
+                        options={Object.values(InspectionType)}
+                        onOptionsChange={onInspectionTypesChange}
+                        label={TranslateText('Inspection type')}
+                        initialSelectedOptions={inspectionTypes}
+                        multiple
+                    ></Autocomplete>
+                    <Search
+                        value={robotName ?? ''}
+                        placeholder={TranslateText('Search for a robot name')}
+                        onChange={onRobotNameChange}
+                    />
+                    <Search
+                        value={tagId ?? ''}
+                        placeholder={TranslateText('Search for a tag')}
+                        onChange={onTagIdChange}
+                    />
+                    <TextField
+                        id="datetime"
+                        value={formattedMinStartTime}
```

Comment:
I tried that first but without success. Need a specific format and since this uses built in date time pickers the format will vary with chosen browser. Didn't find a different alternative but if you have any ideas I would love to try